src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -2620,46 +3547,72 @@ void Instruction::execute() {
         }
         break;
       }
-      case Opcode::AArch64_LD1_MXIPXX_H_S: {  // ld1w {zath.s[ws, #imm]}, pg/z,
-                                              // [<xn|sp>{, xm, LSL #2}]
+      case Opcode::AArch64_LD1_MXIPXX_V_H: {  // ld1h {zatv.h[ws, #imm]}, pg/z,
+                                              // [<xn|sp>{, xm, lsl #1}]
         // SME, LOAD
-        // Not in right context mode. Raise exception
+        // If not in right context mode, raise exception
         if (!ZAenabled) return ZAdisabled();
 
-        const uint16_t partition_num = VL_bits / 32;
+        const uint16_t partition_num = VL_bits / 16;
         const uint32_t ws = sourceValues_[partition_num].get<uint32_t>();
         const uint64_t* pg =
             sourceValues_[partition_num + 1].getAsVector<uint64_t>();
 
         const uint32_t sliceNum =
             (ws + metadata_.operands[0].sme_index.disp) % partition_num;
-        const uint32_t* data = memoryData_[0].getAsVector<uint32_t>();
+        const uint16_t* data = memoryData_[0].getAsVector<uint16_t>();
 
-        uint32_t out[64] = {0};
         for (int i = 0; i < partition_num; i++) {
-          uint64_t shifted_active = 1ull << ((i % 16) * 4);
-          if (pg[i / 16] & shifted_active) {
-            out[i] = data[i];
+          uint16_t* row =
+              const_cast<uint16_t*>(sourceValues_[i].getAsVector<uint16_t>());
+          uint64_t shifted_active = 1ull << ((i % 32) * 2);
+          if (pg[i / 32] & shifted_active) {
+            row[sliceNum] = data[i];
           } else {
-            out[i] = 0;
+            row[sliceNum] = 0;
```

Comment:
This assignment through the pointer to the underlying memory of the registerValue technically breaks our rule (as per the comment) that a registerValue is immutable. This will probably need to be refactored when the reinterpret_cast issue is being resolved with the solution being similar to the implementations below e.g. populate a std::vector and creating in new registerValue

Reply:
Thanks for noticing, will update this to keep RegVals immutable